cvxpy/lin_ops/canon_backend.py

```diff
@@ -1103,6 +1090,305 @@ def _to_dense(x):
+
+        arg_cols = lin.args[0].shape[0] if len(lin.args[0].shape) == 1 else lin.args[0].shape[1]
+
+        if is_param_free_lhs:
+            if len(lin.data.shape) == 1 and arg_cols != lhs.shape[0]:
+                lhs = lhs.T
+            reps = view.rows // lhs.shape[0]
+            stacked_lhs = sp.kron(lhs.T, sp.eye(reps, format="csr"))
+
+            def func(x, _p):
+                return stacked_lhs @ x
+        else:
+            reps = view.rows // next(iter(lhs.values())).shape[-1]
+            lhs_shape = next(iter(lhs.values())).shape
+
+            if len(lin.data.shape) == 1 and arg_cols != lhs_shape[0]:
+                # Example: (n,n) @ (n,), we need to interpret the rhs as a column vector,
+                # but it is a row vector by default, so we need to transpose
+                lhs = {k: v.T for k, v in lhs.items()}
+                lhs_shape = next(iter(lhs.values())).shape
+
+            reps = view.rows // lhs_shape[0]
+            eye = sp.eye(reps, format="csc")
+            stacked_lhs = {k: sp.kron(v.T, eye).tocsc() for k, v in lhs.items()}
+
+            def parametrized_mul(x):
+                return {k: v @ x for k, v in stacked_lhs.items()}
+
+            func = parametrized_mul
+        return view.accumulate_over_variables(func, is_param_free_function=is_param_free_lhs)
+
+    @staticmethod
+    def trace(lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        shape = lin.args[0].shape
+        indices = np.arange(shape[0]) * shape[0] + np.arange(shape[0])
+
+        data = np.ones(len(indices))
+        idx = (np.zeros(len(indices)), indices.astype(int))
+        lhs = sp.csc_matrix((data, idx), shape=(1, np.prod(shape)))
+
+        def func(x, _p):
+            return lhs @ x
+
+        return view.accumulate_over_variables(func, is_param_free_function=True)
+
+    def conv(self, lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        lhs, is_param_free_lhs = self.get_constant_data(lin.data, view, column=False)
+        assert is_param_free_lhs
+        assert lhs.ndim == 2
+
+        if len(lin.data.shape) == 1:
+            lhs = lhs.T
+
+        rows = lin.shape[0]
+        cols = lin.args[0].shape[0]
+        nonzeros = lhs.shape[0]
+
+        lhs = lhs.tocoo()
+        row_idx = (np.tile(lhs.row, cols) + np.repeat(np.arange(cols), nonzeros)).astype(int)
+        col_idx = (np.tile(lhs.col, cols) + np.repeat(np.arange(cols), nonzeros)).astype(int)
+        data = np.tile(lhs.data, cols)
+
+        lhs = sp.csc_matrix((data, (row_idx, col_idx)), shape=(rows, cols))
+
+        def func(x, _p):
+            return lhs @ x
+
+        return view.accumulate_over_variables(func, is_param_free_function=is_param_free_lhs)
+
+    def kron_r(self, lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        lhs, is_param_free_lhs = self.get_constant_data(lin.data, view, column=True)
```

Comment:
I reimplemented these from the scipy backend just to make tests pass, I guess it should be fine to keep them for now, and update it once we tackle their parametrised versions.

Reply:
We will try to add the parametrised version of those functions in a new feature.